plugins/node/opentelemetry-instrumentation-dns/src/dns.ts

```diff
@@ -0,0 +1,210 @@
+            );
+          }
+          return moduleExports;
+        },
+        moduleExports => {
+          if (moduleExports === undefined) return;
+          this._logger.debug('Removing patch for dns');
+          this._unwrap(moduleExports, 'lookup');
+          if (semver.gte(process.version, '10.6.0')) {
+            this._unwrap(moduleExports.promises, 'lookup');
+          }
+        }
+      ),
+    ];
+  }
+
+  /**
+   * Get the patched lookup function
+   */
+  private _getLookup() {
+    return (original: (hostname: string, ...args: unknown[]) => void) => {
+      return this._getPatchLookupFunction(original);
+    };
+  }
+
+  /**
+   * Creates spans for lookup operations, restoring spans' context if applied.
+   */
+  private _getPatchLookupFunction(
+    original: (hostname: string, ...args: unknown[]) => void
+  ) {
+    this._logger.debug('patch lookup function');
+    const plugin = this;
+    return function patchedLookup(
+      this: {},
+      hostname: string,
+      ...args: unknown[]
+    ) {
+      if (
+        utils.isIgnored(hostname, plugin._config.ignoreHostnames, (e: Error) =>
+          plugin._logger.error('caught ignoreHostname error: ', e)
+        )
+      ) {
+        return original.apply(this, [hostname, ...args]);
+      }
+
+      const argsCount = args.length;
+      plugin._logger.debug('wrap lookup callback function and starts span');
+      const name = utils.getOperationName('lookup');
+      const span = plugin._startDnsSpan(name, {
+        attributes: {
+          [AttributeNames.PEER_HOSTNAME]: hostname,
+        },
+      });
+
+      const originalCallback = args[argsCount - 1];
+      if (typeof originalCallback === 'function') {
+        args[argsCount - 1] = plugin._wrapLookupCallback(
+          originalCallback,
+          args[argsCount - 2],
+          span
+        );
+        return plugin._safeExecute(span, () =>
+          (original as LookupFunctionSignature).apply(this, [
+            hostname,
+            ...args,
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          ] as any)
+        );
+      } else {
+        const promise = plugin._safeExecute(span, () =>
+          (original as LookupPromiseSignature).apply(this, [hostname, ...args])
+        );
+        promise.then(
+          result => {
+            utils.setLookupAttributes(span, result as LookupAddress);
+            span.end();
+          },
+          (e: NodeJS.ErrnoException) => {
+            utils.setError(e, span, process.version);
+            span.end();
+          }
+        );
+
+        return promise;
+      }
+    };
+  }
+
+  /**
+   * Start a new span with default attributes and kind
+   */
+  private _startDnsSpan(name: string, options: Omit<SpanOptions, 'kind'>) {
+    return this.tracer
+      .startSpan(name, { ...options, kind: SpanKind.CLIENT })
+      .setAttribute(
+        AttributeNames.COMPONENT,
+        '@opentelemetry/instrumentation-dns'
```

Comment:
nit but i believe you could use `this.instrumentationName`

Reply:
actually attribute component was replaced by named tracers. I will remove it completely